packages/astro/src/runtime/server/util.ts

```diff
@@ -31,3 +31,17 @@ export function serializeListValue(value: any) {
 export function isPromise<T = any>(value: any): value is Promise<T> {
 	return !!value && typeof value === 'object' && typeof value.then === 'function';
 }
+
+export async function* streamAsyncIterator(stream: ReadableStream) {
+	const reader = stream.getReader();
+
+	try {
+		while (true) {
```

Comment:
any code with a `while(true)` gets my approval

Reply:
And a `try / finally`? This code has it all!